File: app/src/main/java/com/kakacat/minitool/textencryption/Contract.java
package com.kakacat.minitool.textencryption;

import com.kakacat.minitool.common.base.IPresenter;
import com.kakacat.minitool.common.base.IView;

public interface Contract {

    interface Presenter extends IPresenter{
        String[] getEncryptionMethods();
        void encryptText(String content,CharSequence method);
    }

    interface View extends IView<Presenter>{
        void onEncryptResult(String decode);
    }
}

File: app/src/main/java/com/kakacat/minitool/textencryption/Contract.kt
package com.kakacat.minitool.textencryption

import com.kakacat.minitool.common.base.IPresenter
import com.kakacat.minitool.common.base.IView

interface Contract {
    interface Presenter : IPresenter {
        val encryptionMethods: Array<String>
        fun encryptText(content: String?, method: CharSequence?)
    }

    interface View : IView<Presenter?> {
        fun onEncryptResult(decode: String?)
    }
}